include/boost/geometry/algorithms/area.hpp

```diff
@@ -287,37 +286,41 @@ struct area
     }
 };
 
-template <BOOST_VARIANT_ENUM_PARAMS(typename T)>
-struct area<boost::variant<BOOST_VARIANT_ENUM_PARAMS(T)> >
+template <typename Geometry>
+struct area<Geometry, dynamic_geometry_tag>
 {
-    typedef boost::variant<BOOST_VARIANT_ENUM_PARAMS(T)> variant_type;
-
     template <typename Strategy>
-    struct visitor
-        : boost::static_visitor<typename area_result<variant_type, Strategy>::type>
+    static inline typename area_result<Geometry, Strategy>::type
+        apply(Geometry const& geometry, Strategy const& strategy)
     {
-        Strategy const& m_strategy;
-
-        visitor(Strategy const& strategy): m_strategy(strategy) {}
-
-        template <typename Geometry>
-        typename area_result<variant_type, Strategy>::type
-        operator()(Geometry const& geometry) const
+        // TODO: Technically value could be returned directly from visit
```

Comment:
Current version looks fine to me and consistent with the version below

Reply:
Plus there is a problem with the old `variant` which forces us to define the return type as the template parameter of `boost::static_visitor<>`. I tried to define it automatically based on the return type of the functor for the first type stored in a variant and it would be doable but the adaptation of `boost::variant` would be a lot more complex. We would be able to write:
```
    return visit<...>(...);
```
but I think in general `area_result<>` would still be needed so I doubt it'd be worth it to make visit more complex only to be able to ommit two lines in some places of the code. I only worry about cutting of the possibility because it's possible that in the (most likely distant) future we could need this for something. But that's only that.